src/terminal/parser/OutputStateMachineEngine.cpp

```diff
@@ -1098,6 +1104,17 @@ bool OutputStateMachineEngine::_GetOscSetClipboard(const std::wstring_view strin
     return SUCCEEDED_LOG(Base64::Decode(substr, content));
 }
 
+// Routine Description:
+// - Takes a sequence id ("final byte") and determines if it accepts sub parameters.
+// Arguments:
+// - id - The sequence id to check for.
+// Return Value:
+// - True, if it accepts sub parameters or else False.
+bool OutputStateMachineEngine::_CanSeqAcceptSubParam(const VTID id) noexcept
+{
+    return id == CsiActionCodes::SGR_SetGraphicsRendition;
+}

```

Comment:
If we want to support sub params in the `DECCARA` and `DECRARA` sequences, this will need to be a little more complicated. In pseudocode, something like this:

```
return id == SGR || (firstSubParam >= 4 && (id == DECCARA || id == DECRARA));
```

Because we want to allow sub params in `DECCARA` and `DECRARA`, but only from parameter 4 onwards, since the first 4 parameters are numeric. I'm hoping it shouldn't be too complicated to track something like `firstSubParam` or an equivalent test.

Reply:
I think, `DECCARA` `DECRARA` is now handled properly. I'll wait for your confirmation.

Reply:
Yeah. This part looks good to me. But see my comments below regarding the location of the `hasSubParamsFor` test.